stdlib/SuiteSparse/test/umfpack.jl

```diff
@@ -189,4 +189,32 @@ using LinearAlgebra: Adjoint, Transpose, SingularException
         end
     end
 
+    @testset "Reuse symbolic LU factorization" begin
+        A1 = sparse(increment!([0,4,1,1,2,2,0,1,2,3,4,4]),
+                    increment!([0,4,0,2,1,2,1,4,3,2,1,2]),
+                    [2.,1.,3.,4.,-1.,-3.,3.,9.,2.,1.,4.,2.], 5, 5)
+        for Tv in (Float64, ComplexF64, Float16, Float32, ComplexF16, ComplexF32)
+            for Ti in Base.uniontypes(SuiteSparse.UMFPACK.UMFITypes)
+                A = convert(SparseMatrixCSC{Tv,Ti}, A0)
+                B = convert(SparseMatrixCSC{Tv,Ti}, A1)
+                b = Tv[8., 45., -3., 3., 19.]
+                F = lu(A)
+                lu!(F, B)
+                @test F\b ≈ B\b ≈ Matrix(B)\b
+
+                # singular matrix
+                C = copy(B)
+                C[4, 3] = Tv(0)
+                F = lu(A)
+                @test_throws SingularException lu!(F, C)
+
+                # change of nonzero pattern
+                D = copy(B)
+                D[5, 1] = Tv(1.0)
+                F = lu(A)
+                @test_throws ArgumentError lu!(F, D)
```

Comment:
From where is this error thrown? I only see that you _set_ the colptrs, rowvals, and nzvals of `F` by those of `S`, but I can't seem to find where you check whether these are consistent with what you precomputed.

Reply:
The error is thrown in line `225`. Because the status code `F.status` is not equal to `UMFPACK_OK`, `issuccess(F)` is false and the error is thrown.